src/os/filestore/FileJournal.cc

```diff
@@ -493,6 +500,10 @@ int FileJournal::open(uint64_t fs_op_seq)
   }
 
   return 0;
+
+out:
+  close();
+  return err;
```

Comment:
I am not sure "::pread(fd, bpdata, bp.length(), 0);" in read_header mustreturn a -EINVAL , it may return -EPERM for non-permitted file or anything else?

I think we can give err a default value with  -EINVAL so that we do not need to assign it everywhere.